domain/objectstore/service/service.go

```diff
@@ -75,13 +86,31 @@ func (s *Service) GetMetadata(ctx context.Context, path string) (objectstore.Met
 	}, nil
 }
 
+// GetMetadataBySHA256 returns the persistence metadata for the object
+// with SHA256 starting with the provided prefix.
+func (s *Service) GetMetadataBySHA256(ctx context.Context, sha256 string) (objectstore.Metadata, error) {
+	if sha256 == "" || !hashRegexp.MatchString(sha256) {
```

Comment:
Those should be 2 separate errors (e.g. can't be empty / not valid)

Reply:
If we commit the above change to `hashRegexp` the value `""` will fail the `MatchString` check anyway. So technically we won't need the `sha246 == ""` check at all.

We still may want to include it separately for the sake of more descriptive errors. But I don't really see a scenerio where someone passes an empty string in here an expects it to work

Reply:
I agree with @jack-w-shaw